controlplane/kubeadm/controllers/controller.go

```diff
@@ -232,6 +232,18 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 	}
 
 	controlPlane := internal.NewControlPlane(cluster, kcp, ownedMachines)
+
+	if requeue := preflightChecks(controlPlane); requeue != nil {
+		return *requeue, nil
+	}
+
+	if result, err := r.reconcileHealth(ctx, cluster, kcp, controlPlane); err != nil {
```

Comment:
this is kind of hard to read considering err and the optional result, but I don't have better ideas...

Reply:
yeah I tend to abuse the multiline if assignments. Probably just using a regular assignment would be less weird looking